### PR TITLE
PP-4610 Fix swagger - Payment & Events API specs

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT YOU DID
Part 1 of fixing swagger file
- Added back fields `language` and `delayed_capture`
- Included missing search refunds API ( GET /v1/refunds)
- Fixed field naming (links -> _links) and nested "payments" key shown in `GET /v1/payments/:id` call
- Corrected specs for following API calls
     - /v1/payments (both GET & POST)
     - /v1/payments/{paymentId}
     - /v1/payments/{paymentId}/events
- All above resulting in new swagger file

**Pending**
- All refund related APIs

### CODE REVIEW

File: src/main/java/uk/gov/pay/api/model/ValidCreatePaymentRequest.java [hunk 1]
```diff
@@ -25,9 +25,9 @@
     private final String description;
     @ApiModelProperty(name = "agreement_id", value = "ID of the agreement being used to collect the payment", required = false, example = "33890b55-b9ea-4e2f-90fd-77ae0e9009e2")
     private String agreementId;
-    @ApiModelProperty(value = "ISO-639-1 Alpha-2 code of a supported language to use on the payment pages", required = false, example = "en")
+    @ApiModelProperty(name = "language", value = "ISO-639-1 Alpha-2 code of a supported language to use on the payment pages", required = false, example = "en", allowableValues = "en,wy")
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/api/model/ValidCreatePaymentRequest.java]: typo? should be 'cy?'

File: src/main/java/uk/gov/pay/api/resources/PaymentsResource.java [hunk 2]
```diff
@@ -92,7 +93,7 @@ public PaymentsResource(Client client,
                     "as 'authorization: Bearer YOUR_API_KEY_HERE'",
             code = 200)
     @ApiResponses(value = {
-            @ApiResponse(code = 200, message = "OK", response = PaymentWithAllLinks.class),
+            @ApiResponse(code = 200, message = "OK", response = PaymentForSearchResult.class),
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/api/resources/PaymentsResource.java]: What's the reason for this change? Do the generated structure for the 2 differ? There's a change to PaymentWithAllLinks in this PR to make the docs generate unwrapped, does this make this unnecessary?

File: src/main/java/uk/gov/pay/api/model/ValidCreatePaymentRequest.java (hunk 1 above)
Comment (kbottla) [reply src/main/java/uk/gov/pay/api/model/ValidCreatePaymentRequest.java]: Good spot 👍. Corrected

File: src/main/java/uk/gov/pay/api/resources/PaymentsResource.java (hunk 2 above)
Comment (kbottla) [reply src/main/java/uk/gov/pay/api/resources/PaymentsResource.java]: Yes, the structures differ. PaymentWithAllLinks doesn't include card details as `Payment` object is used here. PaymentForSearchResult inherits from CardPayment which includes all attributes required.

Changes to PaymentWithAllLinks fixes issues with /v1/payments (POST) endpoint .i.e., unwraps payment object for new payment response
